File: e17/proto/etk_server/examples/etk.py
#!/usr/bin/python

import os
import time

def etk(str) :
    out_file = open("/tmp/etk_server_fifo","w")
    out_file.write("etk_" + str + "\0")
    out_file.close()    
    
    in_file = open("/tmp/etk_server_fifo","r")
    text = in_file.read()
    in_file.close()
    
    return text

os.system("etk_server /tmp/etk_server_fifo &")
time.sleep(0.5)

etk("init")

win = etk("window_new")

button = etk("button_new_with_label \"Python rules!\"")

etk("container_add " + win + " " + button)
etk("window_title_set " + win + " \"Etk-Python Demo\"")
etk("widget_show_all " + win)

etk("server_signal_connect \"clicked\" " + button + " \"button_click\"")
etk("server_signal_connect \"delete_event\" " + win + " \"window_delete\"")

while 1:
    
    event = etk("server_callback")
    
    if event == "window_delete":
	print "Bye!"
	break    
    
    if event == "button_click":
	print "Button clicked!"
    
etk("main_quit")
etk("server_shutdown")

File: e17/proto/etk_server/examples/etk.rb
#!/usr/local/bin/ruby

puts("Depricated for now!\n")
exit

def etk(str)

  fifo = File.open("/tmp/etk_server_fifo", "w")
  fifo.puts("etk_#{str}")
  fifo.close()
  
  fifo = File.open("/tmp/etk_server_fifo", "r")
  ret = fifo.gets()
  fifo.close()
  
  return ret
end

system("etk_server /tmp/etk_server_fifo &")
sleep(0.5)

etk("init")

win = etk("window_new")
etk("window_title_set #{win} \"Etk-Ruby Demo\"");

vbox = etk("vbox_new 0 0")
etk("container_add #{win} #{vbox}")

button1 = etk("button_new_with_label \"Ruby rules!\"")
etk("box_pack_start #{vbox} #{button1} 1 1 0")

button2 = etk("button_new_with_label \"So does Etk!\"")
etk("box_pack_start #{vbox} #{button2} 1 1 0")

entry = etk("entry_new")
etk("box_pack_start #{vbox} #{entry} 1 1 0")

button3 = etk("button_new_with_label \"Get text\"")
etk("box_pack_start #{vbox} #{button3} 1 1 0")

etk("widget_show_all #{win}")

etk("server_signal_connect \"clicked\" #{button1} \"button_1_click\"")
etk("server_signal_connect \"clicked\" #{button2} \"button_2_click\"")
etk("server_signal_connect \"clicked\" #{button3} \"button_3_click\"")
etk("server_signal_connect \"delete_event\" #{win} \"win_delete\"")

event = ""
while event != "win_delete"

  event = etk("server_callback");
  case event
  
    when "button_1_click"
      puts "button 1 clicked"
  
    when "button_2_click"
      puts "button 2 clicked"
     
    when "button_3_click"
      text = etk("entry_text_get #{entry}")
      text = etk("server_var_get #{text}")
      puts "Text=#{text}"
     
   end
  
end

etk("main_quit");
etk("shutdown");
etk("server_shutdown");
